sql/2020/06_Third_Parties/tao_by_third_party.sql

```diff
@@ -0,0 +1,78 @@
+#standardSQL
+# Percent of third-party requests with "Timing-Allow-Origin" headers
+# Header reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Timing-Allow-Origin 
+
+CREATE TEMP FUNCTION get_tao(headers STRING)
+RETURNS STRING LANGUAGE js AS '''
+  try {
+    const regex = /timing-allow-origin = (\\*|(http.*?,? )+)/gm;
```

Comment:
This seems brittle. Are you getting the results you'd expect? If not let's think of alternate approaches. For example, the `requests` dataset contains key/value pairs of response headers, which may be more consistent to parse.

Reply:
Yes, did a couple of approaches and this one seems to work fine with cases:
- wildcard
- single domain
- multiple domains

My intention was to make a cheaper query, using `summary_requests `instead of `requests`.
And yeah, It would be easier if `respOtherHeaders` contained JSON data, instead of comma-separated.

Reply:
Sounds good. Resolving.